### DISCUSSION TITLE
How to test datasets of our own

### DISCUSSION THREAD

**Discussion by Kaisor-Yuan** [IRVLUTD/Proto-CLIP#12] [Q&A] [answered]
If we want to test Proto-CLIP in our dataset,how can we adapt this code?

[search_scale, search_step]how to fix these two parameters

**Comment by jishnujayakumar** [comment, marked as answer]
Hi @dhhxhhh,

For any dataset other than the ones listed in [Proto-CLIP datasets](https://github.com/IRVLUTD/Proto-CLIP/blob/main/datasets/), you need to 

1. Download and place the dataset in the `DATA/` directory.
2. Create one dataset-specific parsing file inside https://github.com/IRVLUTD/Proto-CLIP/blob/main/datasets/
3. Create an alias for it inside https://github.com/IRVLUTD/Proto-CLIP/blob/main/datasets/__init__.py
4. Create a config for it and place it inside https://github.com/IRVLUTD/Proto-CLIP/tree/main/configs
5. [search_scale, search_step] are not very important as they are mainly meant to form the {**alpha**, **beta**} grid search values. You can tune it as per your need.
6. {**alpha**, **beta**} are critical hyperparameters. As this work uses a grid search to find the appropriate {**alpha**, **beta**} for each new dataset and CLIP backbone pair, please use it for the new dataset. 

I hope that these instructions would help you to conduct experiments on Proto-CLIP with your intended dataset.

**Reply by Kaisor-Yuan** [reply]
I got it. Thank you!
